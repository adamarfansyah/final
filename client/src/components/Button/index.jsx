import PropTypes from 'prop-types';
import classNames from 'classnames';
import classes from './style.module.scss';

const Button = ({ children, onClick, className, disabled, variant }) => {
  const buttonClasses = classNames(classes.btn, className, {
    [classes.btnSecondary]: variant === 'secondary',
  });

  return (
    <button data-testid="btn" className={buttonClasses} onClick={onClick} type="submit" disabled={disabled}>
      {children}
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func,
  className: PropTypes.string,
  disabled: PropTypes.bool,
  variant: PropTypes.oneOf(['primary', 'secondary']),
};

Button.defaultProps = {
  disabled: false,
  variant: 'primary',
};

export default Button;
