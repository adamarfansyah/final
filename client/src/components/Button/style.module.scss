@import '@styles/base';

.btn {
  background-color: var(--color-button);
  color: white;
  padding: 10px 20px;
  border: none;
  border-radius: 0.8rem;
  font-weight: bold;
  cursor: pointer;
  &:hover {
    background-color: $color-green-md;
  }
}

.btnSecondary {
  background-color: transparent;
  border: 2px solid var(--color-button);
  color: var(--color-button);
  &:hover {
    background-color: var(--color-button);
    color: white;
  }
  &:disabled {
    background-color: rgb(79, 79, 79);
    cursor: not-allowed;
    color: rgb(145, 145, 145);
    border: none;
  }
}
