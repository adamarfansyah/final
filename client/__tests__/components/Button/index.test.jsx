import { render, fireEvent } from '@testing-library/react';
import Button from '@components/Button';

describe('Component Button', () => {
  test('Should render correctly with default variant', () => {
    const { getByText } = render(<Button>Click me</Button>);
    const button = getByText('Click me');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('btn');
  });

  test('Should render correctly with secondary variant', () => {
    const { getByText } = render(<Button variant="secondary">Click me</Button>);
    const button = getByText('Click me');
    expect(button).toBeInTheDocument();
    expect(button).toHaveClass('btnSecondary');
  });

  test('Should call onClick prop when button is clicked', () => {
    const onClickMock = jest.fn();
    const { getByText } = render(<Button onClick={onClickMock}>Click me</Button>);
    const button = getByText('Click me');
    fireEvent.click(button);
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });

  test('Should be disabled when disabled prop is true', () => {
    const { getByText } = render(<Button disabled>Click me</Button>);
    const button = getByText('Click me');
    expect(button).toBeDisabled();
  });

  test('Should match with snapshot', () => {
    const { asFragment } = render(<Button>Click me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
