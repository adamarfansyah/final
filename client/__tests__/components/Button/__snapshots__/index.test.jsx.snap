// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button Should match with snapshot 1`] = `
<DocumentFragment>
  <button
    class="btn"
    data-testid="btn"
    type="submit"
  >
    Click me
  </button>
</DocumentFragment>
`;
